### ISSUE TITLE
About reproducing the Macro F1 score on SemEval2018

### ISSUE THREAD

**Issue by vhientran** [FFTYYY/RoR_relation_extraction#7]
Hi @zhijing-jin ,

Sorry for disturbing you, but I already tried to run your source code as guided. However, on SemEval2018, I obtain Macro F1 score is 35.9% (the best result), whereas the reported result is 38.83% with RoR_base. I use the same auto hyper-parameters and the same environment configuration (as in requirement.txt). What should I do to get the reported performance? Thanks so much!

**Comment by zhijing-jin** [comment]
Hi, thank you for your question. We have reproduced our results on multiple different servers.

Nonetheless, it is still possible that you find that the current hyperparameter set does not generate the same results. I would say that trying the random seed `--t2g_seed` would probably help (although in our experiments we fix the random seed which we set very casually for all settings). Other hyper parameters that worth trying include the class weights of different relations, learning rate `--t2g_lr`, and some common ones.

I will also involve the co-first author Yongyi @FFTYYY in the discussion in case he has some other suggestions.

**Comment by FFTYYY** [comment]
thank you for your attention to our work. I reran the experiment of RoR_base just now on a different mechine, I got macro f1=40.23, the command I use is 
``` CUDA_VISIBLE_DEVICES=0,1,2,3 python main.py --ensemble=1 --epoch_numb=30 --no_rel_name=NONE --valid_text=./data/semeval_2018_task7/2.test.text.xml --valid_rels=./data/semeval_2018_task7/keys.test.2.txt --loss=loss_2 --no_valid --warmup_prop=0.1 --scheduler=cosine --t2g_batch_size=8 --t2g_lr=1e-4```
you can check if you used the same hyperparameters.

also it's worth noting that although we fixed random seed in our code, the results can vary on different mechines due to some mysterious randomness of cuda. and the standard deviation of semeval dataset seems to be relatively large, so sometimes you can get a somewhat low result. I recommand you try different random seed. and the ensembled version should be more stable.

**Comment by vhientran** [comment]
Thank you very much for your quick response and your help. 
I will try to fine-tune hyper-parameters and use different random seeds, as you suggested.
